### DISCUSSION TITLE
.editorconfig issue

### DISCUSSION THREAD

**Discussion by kachnitata** [BlueMystical/Dark-Mode-Forms#21] [Q&A]
Hi,
(not sure whether to ask this questoin here or in the "Issues" section)

I tried to use Ctrl-K Ctrl-D in Visual Studio to reformat according to the current .editorconfig (provided by @TetraTheta )
While it correctly fixed indentation in the parts that I had edited, it also introduced other changes like

```
-			public Rectangle ToRectangle()
+			public Rectangle ToRectangle ()

```
and so on.
I think it is because the current .editorconfig contains

`csharp_space_between_method_declaration_name_and_open_parenthesis = true
`
but surprisingly, the (presumably formatted) source files found here in the repository seem to not obey this setting.

Any ideas? I personally prefer to set it to false but more important question is why the setting was ignored when the files were being formatted the first time but not the second time...

**Comment by TetraTheta** [comment]
I think, in general, Visual Studio is not good at supporting `.editorconfig`. Even though VS said it loaded `.editorconfig`, the configuration isn't correctly applied.  
Visual Studio's own formatting rule will be always applied, though. If you want correct formatting no matter what, you'll have to configure your VS to match configuration of `.editorconfig` one by one. Yes, this sucks.

This is why I use CodeMaid for formatting stuff. But CodeMaid also have few problems with formatting, so you'll have to manualy format your code by hand sometimes.

---

This is just my personal opinion, so think nothing of this.

There is no good, free alternative IDEs of Visual Studio Community Edition for C# project.

JetBrain Rider would be good alternative, but it is paid software. If you're a student, you can use it with Free Educational Licenses, though.  
Visual Studio Code is merely rich text editor to me. You can install lots of extensions, but I've found that it will slow down initial loading speed of VSC a lot. Not to mention that it lacks GUI Designer feature.

And there is no other free IDE that supports GUI Designer feature as far as I know.

So... I think we must just bear with it.

**Comment by kachnitata** [comment]
>I think, in general, Visual Studio is not good at supporting .editorconfig.

But regarding this particular issue - it seems that my VS _did_ respect ` .editorconfig` but whatever tool formatted the files before  (I suppose it was done by you or perhaps @BlueMystical  ) ignored it... 

for this particular setting - changing to 

`csharp_space_between_method_declaration_name_and_open_parenthesis = false
`
would in fact make the source files comply with ` .editorconfig` , without any need to modify them.
So If no one objects, I am going to create a PR for that change and we are done.

**Reply by TetraTheta** [reply]
IIRC, `csharp_space_between_method_declaration_name_and_open_parenthesis` is `false` by default in VS. That would be why changing it to `false` will make VS look like complying with `.editorconfig`.

I forgot to say that the `.editorconfig` override doesn't always happen. Sometimes, it will work. But also, it won't work in other times. I've  experienced it a lot, and that's why I tried to find out alternative IDEs of VS.

This kind of discussion would never happen if VS overrided its own rule correctly... *\*sigh\**

**Reply by kachnitata** [reply]
I may sound stubborn and annoying, but I really don't think there is any fault on VS's side.
If I understand correctly that the repository has been reconstructed and reformatted with CodeMaid, then that's probably where the current discrepancy has originated. When I tried to re-format the files the VS worked as expected. Both true and false led to the respective result. I just proposed setting the flag to false to reflect the current state of the source files.

>. That would be why changing it to false will make VS look like complying with .editorconfig.

I do not agree... That would not _look like_ but would re-align the `*.cs` files and `.editorconfig`

**Reply by kachnitata** [reply]
But on the other hand, if the repository reconstruction has been done in pure VS then of course I am wrong about there being no fault on VS's side.

**Reply by TetraTheta** [reply]
You may be right.

I don't use `.editorconfig` generated by VS as-is. I always use _alphabetically sorted_ `.editorconfig`, without any comment, like [this](https://github.com/TetraTheta/Small-Projects/blob/main/ConvertWebP/.editorconfig). Maybe that's the reason of those behavior.  
Also, my Windows installation environment is messed up. This might affect VS too.

But I checked that VS recognized `.editorconfig` file and showed me GUI configuration tab for it. There should be no problem with code formatting style.

All I can tell is, at least in my environment, VS doesn't format codes as I expected, even with CodeMaid. I have to manually format those parts.

**Reply by kachnitata** [reply]
Who knows, it is quite complicated...
Because I use VS with Reshaper (R#), it may affect things, too. So while I was saying my VS worked as expected, it might pretty well be because R# was helping it behind the scenes...

I will stay alert and keep in mind your comment about manually changing each one of the settings to match on both sides (VS and .editorconfig). I hope I will not have to do so, because I use the same VS instance with other projects.

**Comment by BlueMystical** [comment]
after the Re-structure i only touched the Form1 for examples, in which the only format i made was by clicking the 'Format all Document' button on VS.
